skopt/optimizer/base.py

```diff
@@ -17,7 +18,67 @@
 from ..utils import eval_callbacks
 
 
+class _WrapFuncNamedArgs:
```

Comment:
Instead of creating (and having to maintain) our own function wrapper we could use [`functools.wraps`](https://docs.python.org/2/library/functools.html#functools.wraps) like so:
```python
In [3]: def adaptor(f):
   ...:     @wraps(f)
   ...:     def wrapper(*args):
   ...:         print('modify args', args)
   ...:         return f(*args)
   ...:     return wrapper
   ...: 

In [4]: def my_objective(x, y, z):
   ...:     """doc here"""
   ...:     return x + y + z
   ...: 

In [5]: adapted_obj = adaptor(my_objective)

In [6]: adapted_obj?
Signature: adapted_obj(x, y, z)
Docstring: doc here
File:      ~/Downloads/<ipython-input-4-84edd263a788>
Type:      function
```
it takes care of forwarding doc strings, and signatures, etc.

Reply:
Thanks for the tip, I'll take a look.